### PR TITLE
Re-writing the sample.

### PR DESCRIPTION
**Pull request by jonparrott**
* Changing auth to usable auth piggybacking on gcloud credentials.
* Providing succinct examples for listing, insterting, and deleting instances.
* Following current convention of main() and run() functions.
* Adding markers for direct inclusion into cloud docs.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2015 Google, Inc
```
Comment (tmatsuo) [inline LICENSE]: This is not correct. Please keep the brackets as they are.

File: README.md [hunk 2]
```diff
@@ -1,55 +1,54 @@
...
-- Update the client_secrets.json file with your client id and secret found in
-  the [Google APIs Console][3].
-- Run the sample using the command:
+## Pre-requisites
 
+1. Create a project on the [Google Developers Console](https://console.developers.google.com) and [enable billing](https://console.developers.google.com/project/_/settings).
+2. Install the [Google Cloud SDK](https://cloud.google.com/sdk/)
+
+```bash
+curl https://sdk.cloud.google.com | bash
```
Comment (tmatsuo) [inline README.md]: curl and bash combination is controvertial...

File: main.py [hunk 3]
```diff
@@ -12,158 +12,153 @@
...
-INSTANCE_NAME = 'startup-script-demo'
-DISK_NAME = INSTANCE_NAME + '-disk'
-INSERT_ERROR = 'Error inserting %(name)s.'
-DELETE_ERROR = """
-Error deleting %(name)s. %(name)s might still exist; You can use
-the console (http://cloud.google.com/console) to delete %(name)s.
+Google Compute Engine demo using the Google Python Client Library.
 """
 
+__author__ = 'jonwayne@google.com (Jon Wayne Parrott)'
```
Comment (tmatsuo) [inline main.py]: Can you use 4 spaces indent?

File: main.py [hunk 4]
```diff
@@ -12,158 +12,153 @@
...
+    zone=zone,
+    instance=name).execute()
+# [END delete_instance]
+
+
+# [START wait_for_operation]
+def wait_for_operation(compute, project, zone, operation):
+  sys.stdout.write('Waiting for operation to finish')
+  while True:
+    result = compute.zoneOperations().get(project=project, zone=zone, operation=operation).execute()
```
Comment (tmatsuo) [inline main.py]: long line maybe?

**Comment by tmatsuo** [comment]
I just put some comments

**Comment by tmatsuo** [comment]
How about to put travis config and a simple tox configuration, just to check coding style?

File: README.md (hunk 2 above)
Comment (jonparrott) [inline README.md]: Yes, but unresolved at this point. We shouldn't let that block the sample is it's the exact same instructions that are on the gcloud landing page.

File: main.py (hunk 4 above)
Comment (jonparrott) [inline main.py]: fixed.

File: main.py (hunk 3 above)
Comment (jonparrott) [inline main.py]: done.

File: LICENSE (hunk 1 above)
Comment (jonparrott) [inline LICENSE]: fixed.

**Comment by jonparrott** [comment]
Added travis and tox config and make sure everything is passing. @tmatsuo any other tweaks?

File: README.md (hunk 2 above)
Comment (tmatsuo) [inline README.md]: I still don't like it...

How do you track when the instructions on the gcloud landing page and change the instructions here too?

File: README.md [hunk 5]
```diff
@@ -1,55 +1,54 @@
...
+gcloud auth login
+gcloud config set project your-project-id
+```
+3. Install dependencies using [pip](https://pypi.python.org/pypi/pip)
+
+```bash
+pip install -r requirements.txt
```
Comment (tmatsuo) [inline README.md]: Does pip try to install things to user's directory by default? I mean, if pip tries to install things in a directory where the user can not write, they'll get a permission error.

**Comment by tmatsuo** [comment]
LGTM
I still have some questions, but you can merge.

File: README.md (hunk 5 above)
Comment (jonparrott) [inline README.md]: It varies by OS, distribution, and user settings. I think we can give python developers the benefit of the doubt and assume they know how to use their package manager.

File: README.md (hunk 2 above)
Comment (jonparrott) [inline README.md]: I don't know. There's an ongoing discussion about this, but the discussion is more around how to let users verify the download first. This method of installing gcloud is used in multiple samples and shouldn't be too hard to grep if we need to update at any point.

**Comment by jonparrott** [comment]
@elibixby I don't have collaborator access to this repo. Can you merge?

**Pull request merged**